File: DSA/SuryanshMishra_590012069/Day9/day9question3.py
def first_non_repeating_char(s):
    freq = {}
    
    # Count frequency of each character
    for c in s:
        freq[c] = freq.get(c, 0) + 1
    
    # Find first character with frequency 1
    for c in s:
        if freq[c] == 1:
            return c
    
    return None

def main_non_repeating():
    s = input("Enter a string: ")
    result = first_non_repeating_char(s)
    
    if result is None:
        print("No non-repeating character found.")
    else:
        print(f"First non-repeating character: {result}")

File: DSA/PriyankaSingh_590016827/day9question3.c
#include <stdio.h>
#include <limits.h>
#include <string.h>

#define CHAR_RANGE 256

int firstNonRepeatingChar(const char *str) 
{
    int frequency[CHAR_RANGE] = {0};
    int len = strlen(str);
    
    // Building frequency count
    for (int i = 0; i < len; i++) 
    {
        frequency[(unsigned char)str[i]]++;
    }
    
    // Finding first character with frequency 1
    for (int i = 0; i < len; i++) 
    {
        if (frequency[(unsigned char)str[i]] == 1) 
        {
            return str[i];
        }
    }
    
    return -1; // No non-repeating character found
}

int main() 
{
    char input[1000];
    
    printf("Enter a string: ");
    fgets(input, sizeof(input), stdin);
    
    // Removing newline character if present
    input[strcspn(input, "\n")] = '\0';
    
    int result = firstNonRepeatingChar(input);
    
    if (result == -1) 
    {
        printf("All characters repeat or string is empty\n");
    } 
    else 
    {
        printf("First non-repeating character: '%c'\n", result);
    }
    
    return 0;
}